core/src/main/java/org/infinispan/interceptors/impl/CacheLoaderInterceptor.java

```diff
@@ -265,8 +241,8 @@ public Object visitKeySetCommand(InvocationContext ctx, KeySetCommand command)
             return rv;
          }
 
-         CacheSet<K> keySet = (CacheSet<K>) rv;
```

Comment:
You can remove the REMOTE_ITERATION optimization above for the KeySetCommand as well and the value in the FlagBitSets now that we don't use an iterator.